main/main.cpp

```diff
@@ -479,13 +487,19 @@ void Main::test_cleanup() {
 	ResourceSaver::remove_custom_savers();
 
 #ifdef TOOLS_ENABLED
+	NativeExtensionManager::get_singleton()->deinitialize_extensions(NativeExtension::INITIALIZATION_LEVEL_EDITOR);
+	uninitialize_modules(MODULE_INITIALIZATION_LEVEL_EDITOR);
```

Comment:
Inconsistency: `deinitialize` / `uninitialize`. I think the latter is correct, but not sure.

Reply:
I feel we should rename stuff in GDExtension from deinitialize to uninitialize, but it likely should be done in a subsequent PR.

Reply:
According to Google:
- "deinitialize" 51700 results
- "uninitialize" 64000 results
Kinda seems the later is a bit more popular.

Reply:
For fun, but I guess we should probably ensure a single term for the whole engine - https://twitter.com/reduzio/status/1521782429526200321

Reply:
I think `finalize` as proposed in your poll might be a better term. `uninitialize` might be a bit more common than `deinitialize` but it's also used for _uninitialized variable_, which means something which _has not been initialized yet_. While here things have been initialized, used, and now need to be cleaned up (which is usually different from just de-initializing variables).

Actually as I write this, "deinitialize" sounds a bit better to avoid the confusion with _not yet initialized_ "uninitialized". That is if we want to keep "initialize" as part of the name since the enums are about `INITIALIZATION_LEVEL`. Otherwise I think another term like "finalize" can be clearer and less prone to typos (it's always super difficult to spell uninitialize IMO with all those ni's :P).

Reply:
We can discuss this with other contributors I guess, but anything we decide we should probably make it consistent across the codebase.